src/tiled/scriptfileformatwrappers.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * scriptfileformatwrappers.h
+ * Copyright 2019, Thorbjørn Lindeijer <bjorn@lindeijer.nl>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <QObject>
+
+namespace Tiled {
+
+class EditableTileset;
+class EditableMap;
+class MapFormat;
+class TilesetFormat;
+
+class ScriptTilesetFormatWrapper : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY(QString name READ name)
+    Q_PROPERTY(QString extension READ extension)
+
+public:
+    explicit ScriptTilesetFormatWrapper(TilesetFormat *format, QObject *parent = nullptr);
+
+    QString name() const;
+    QString extension() const;
+
+    Q_INVOKABLE EditableTileset *read(QString &filename);
+    Q_INVOKABLE QString write(EditableTileset *tileset, QString &filename);
+    Q_INVOKABLE bool supportsFile(QString &filename) const;
+
+private:
+    TilesetFormat *mFormat;
+};
+
+class ScriptMapFormatWrapper : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY(QString name READ name)
+    Q_PROPERTY(QString extension READ extension)
+
+public:
+    Q_INVOKABLE ScriptMapFormatWrapper();
+    Q_INVOKABLE explicit ScriptMapFormatWrapper(MapFormat *format, QObject *parent = nullptr);
```

Comment:
Why are these constructors `Q_INVOKABLE`? Should not be needed and the default constructor can be removed as well.

Reply:
That was a remnant of a test I did, I'll definitely remove it.

I was trying to figure out why it wouldn't accept the return type, and as part of that I tried making a qMetaObject entry for it and then constructed it via JS.